sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableConfigTranslator.java

```diff
@@ -0,0 +1,342 @@
+      builder.setInstanceId(ValueProvider.StaticValueProvider.of(options.getInstanceId()));
+    }
+
+    if (options.getAppProfileId() != null && config.getAppProfileId() == null) {
+      builder.setAppProfileId(ValueProvider.StaticValueProvider.of(options.getAppProfileId()));
+    }
+
+    if (!options.getDataHost().equals("bigtable.googleapis.com")
+        && config.getEmulatorHost() == null) {
+      builder.setEmulatorHost(String.format("%s:%s", options.getDataHost(), options.getPort()));
+    }
+    // - instance of credentials, P12, SUPPLIED, suppliedjson - credentails
+    // - default credentials , needs to be called on the worker GoogleCredentialProvider
+
+    if (options.getCredentialOptions() != null) {
+      try {
+        CredentialOptions credOptions = options.getCredentialOptions();
+        switch (credOptions.getCredentialType()) {
+          case DefaultCredentials:
+            // Veneer uses GoogleDefaultCredentials, so we don't need to reset it
+            break;
+          case P12:
+            String keyFile = ((CredentialOptions.P12CredentialOptions) credOptions).getKeyFile();
+            String serviceAccount =
+                ((CredentialOptions.P12CredentialOptions) credOptions).getServiceAccount();
+            try {
+              KeyStore keyStore = KeyStore.getInstance("PKCS12");
+
+              try (FileInputStream fin = new FileInputStream(keyFile)) {
+                keyStore.load(fin, "notasecret".toCharArray());
+              }
+              PrivateKey privateKey =
+                  (PrivateKey) keyStore.getKey("privatekey", "notasecret".toCharArray());
+
+              if (privateKey == null) {
+                throw new IllegalStateException("private key cannot be null");
+              }
+              builder.setCredentialsProvider(
+                  FixedCredentialsProvider.create(
+                      ServiceAccountJwtAccessCredentials.newBuilder()
+                          .setClientEmail(serviceAccount)
+                          .setPrivateKey(privateKey)
+                          .build()));
+            } catch (GeneralSecurityException exception) {
+              throw new RuntimeException("exception while retrieving credentials", exception);
+            }
+            break;
+          case SuppliedCredentials:
+            Credentials credentials =
+                ((CredentialOptions.UserSuppliedCredentialOptions) credOptions).getCredential();
+            builder.setCredentialsProvider(FixedCredentialsProvider.create(credentials));
+            break;
+          case SuppliedJson:
+            CredentialOptions.JsonCredentialsOptions jsonCredentialsOptions =
+                (CredentialOptions.JsonCredentialsOptions) credOptions;
+            synchronized (jsonCredentialsOptions) {
+              if (jsonCredentialsOptions.getCachedCredentials() == null) {
+                jsonCredentialsOptions.setCachedCredentails(
```

Comment:
I dont think the CachedCredentials are relevant here...just 
                    GoogleCredentials.fromStream(jsonCredentialsOptions.getInputStream()));